src/warc2zim/main.py

```diff
@@ -90,39 +98,30 @@ def get_data(self):
 
 
 # ============================================================================
-class WARCPayloadArticle(BaseWARCArticle):
+class WARCPayloadArticle(BaseArticle):
     """ WARCPayloadArticle used to store the WARC payload
     Usually stored under A namespace
     """
 
     def __init__(self, record):
-        super(WARCPayloadArticle, self).__init__(record)
+        super(WARCPayloadArticle, self).__init__()
+        self.record = record
         self.url = record.rec_headers.get("WARC-Target-URI")
-        self.mime = self._compute_mime()
-        # TODO: converting text/html to text/unchanged-html to avoid rewriting by kiwix
-        # original mime type still preserved in the headers block
-        if self.mime:
-            self.mime = self.mime.split(";", 1)[0]
-            if self.mime == "text/html":
-                self.mime = "text/unchanged-html"
-        else:
-            self.mime = "application/octet-stream"
-
+        self.mime = get_record_mime_type(record)
+        self.title = self.url
         self.payload = self.record.content_stream().read()
 
-    def _compute_mime(self):
-        if self.record.http_headers:
-            # if the record has HTTP headers, use the Content-Type from those (eg. 'response' record)
-            return self.record.http_headers["Content-Type"]
-
-        # otherwise, use the Content-Type from WARC headers
-        return self.record.rec_headers["Content-Type"]
+        # TODO: converting text/html to application/octet-stream to avoid rewriting by kiwix
+        # original mime type still preserved in the headers block
+        if self.mime == "text/html":
+            self.mime = "application/octet-stream"
+            self.title = parse_title(self.payload) or self.url
```

Comment:
I think we should fallback to `""`. That's how we do elsewhere. If source page has no title, then it's not findable via suggest.